File: C/Searching/linear_search.c
#include<stdio.h>

int ar_search(int *arr, int val) 
{
	int len = sizeof(arr)/sizeof(arr[0]);
	for(int i = 0; i < len; i++) {
		if(arr[i] == val) {
			return i;
		}
	}
	return -1; 
}

int main()
{
	
	int arr[] = {10, 20, 40, 65, 39, 44};
	printf("enter the value to search: ");
	int e;
	scanf("%d", &e);
	
	int got = ar_search(arr, e);
	if(got == -1) {
		printf("Unable to find the given element\n");
	}
	else {
		printf("Got the element at position %d", got); 
	}
}

File: Python/Searching/linear_search.py
# Author: Shaikh Aquib
# Simple Linear Search Algorithm
# Time complexity is O(n)

def search(arr, to_search):
    for i in range(0, len(arr)):
        if arr[i] == to_search:
            return i
    return -1

if __name__ == "__main__":
    arr = [10,20,30,44,66,10]    
    to_search = int(input("Enter an element to search: "))
    got = search(arr, to_search)
    if got == -1:
        print("Unable to find the value: ")
    else:
        print("Got the element at position: {}".format(got))



